### ISSUE TITLE
[Ruby]: Add SQL Injection Sinks

### ISSUE THREAD

**Issue by maikypedia** [github/securitylab#740]
### Query PR

https://github.com/github/codeql/pull/12832
https://github.com/github/codeql/pull/13062
https://github.com/github/codeql/pull/13066

### Language

Ruby

### CVE(s) ID list

- [CVE-2014-3482](https://github.com/advisories/GHSA-mhwp-qhpc-h3jm)

### CWE

CWE-089: SQL Injection

### Report

This covers Pg (ruby library for interacting with PostgreSQL databases) SQL Injection vulnerability, that happens when an attacker is able to inject SQL code into the query construction which gets executed : `exec`, `async_exec`, `exec_params`, `async_exec_params` and `prepare`/`exec_prepared`.

The dataflow configuration I used is the SQL Injection default, looking for `RemoteFlowSource` flowing to the construction of queries by Pg. The query also enforce that the former construction gets executed (`prepare`/`exec_prepared`).

Update: I have included `mysql2` and `sequel` libraries to the submission, certain methods in these libraries are potentially dangerous as they allow attackers to perform SQL injection attacks since them executes raw SQL statements. To avoid false positives, I have also included `escape`(mysql2) and `quote`(sqlite3) as sanitizers.

### Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc).

- [X] Yes
- [ ] No

**Comment by ghsecuritylab** [comment]
Your submission is now in status **Final decision**.

_For information, the evaluation workflow is the following:
Initial triage > Test run > Results analysis > Query review > Final decision > Pay > Closed_

**Comment by xcorail** [comment]
Hi maikypedia,

Thanks for the submission! We have reviewed your report and validated your findings. After internally assessing the findings and the query we have determined this query is not actually covering the CVE, and not finding any new true positive, and is therefore not eligible for a reward under the Bug Bounty program.

Even though this submission was assessed as being ineligible, we think that your contribution is valuable for the community as it models sinks in popular Ruby SQL libraries, and we appreciate the effort you put into it. Therefore we would still like to offer you a $500 reward.

Best regards and happy hacking!
